lxd/storage/backend_lxd.go

```diff
@@ -1255,10 +1271,19 @@ func (b *lxdBackend) MountInstance(inst instance.Instance, op *operations.Operat
 		return false, err
 	}
 
-	// Get the volume name on storage.
+	// Get the root disk device config.
+	_, rootDiskConf, err := shared.GetRootDiskDevice(inst.ExpandedDevices().CloneNative())
```

Comment:
Picking a random one of those as an example, there are many more including a bunch of pre-existing ones.

If the intent for the config in `newVolume` is to pass it a volume configuration (`api.StorageVolume.Config`), then this isn't doing it at all. What we're passing here is a `disk` device config from an Instance. They both have a `size` key which is probably all we cared about so far, but that's about it as far as commonality.

I suspect we instead want the DB state of the actual storage volume, which should have the matching `size` property we care about and will also have the rest of the volume configuration including driver specific configuration.

@tomponline

Reply:
Sorted this on IRC, @tomponline will send a separate branch which gets us the behavior we want, though this branch's behavior is no more broken than what we have right now, so fine to merge.